snmp/tests/conftest.py

```diff
@@ -43,6 +51,87 @@ def dd_environment():
                     output.write(response.content)
 
         with docker_run(os.path.join(COMPOSE_DIR, 'docker-compose.yaml'), env_vars=env, log_patterns="Listening at"):
-            yield generate_container_instance_config(
-                SCALAR_OBJECTS + SCALAR_OBJECTS_WITH_TAGS + TABULAR_OBJECTS
-            ), E2E_METADATA
+            if AUTODISCOVERY_TYPE == 'agent':
+                instance_config = {}
+                e2e_metadata['docker_volumes'] = [
+                    '{}:/etc/datadog-agent/datadog.yaml'.format(create_datadog_conf_file(tmp_dir))
+                ]
+            else:
+                instance_config = generate_container_instance_config(
+                    SCALAR_OBJECTS + SCALAR_OBJECTS_WITH_TAGS + TABULAR_OBJECTS
+                )
+            yield instance_config, E2E_METADATA
```

Comment:
Shouldn't it be `e2e_metadata` which is `E2E_METADATA` updated when using agent autodiscovery?

Reply:
Nice catch. It was working because we update the global `E2E_METADATA`. See fix in:
- https://github.com/DataDog/integrations-core/pull/7145/commits/0a7be230ce87e71a101c7d5e847d83ef714259e6
- https://github.com/DataDog/integrations-core/pull/7145/commits/91d9fc3f7a7b55eda4e83491e652ba1b12ef1e46